### ISSUE TITLE
no projects configured

### ISSUE THREAD

**Issue by urban-it-solutions** [francescou/docker-compose-ui#15]
tried to build Container from scratch

output:

Projects

{{id}}
filter by project name:  
no projects configured, please run the container with the option
-v /home/user/docker-compose-ui/demo-projects:/opt/docker-compose-projects:ro
Reload projects Create new project Settings

Logs are looking good:
DEBUG:root:{'node-redis': '/opt/docker-compose-projects/node-redis', 'volumes-demo': '/opt/docker-compose-projects/volumes-demo', 'hello-node': '/opt/docker-compose-projects/hello-node'}
WARNING:werkzeug: \* Debugger is active!
INFO:werkzeug: \* Debugger pin code: 121-154-210
INFO:werkzeug:172.20.82.232 - - [06/Jan/2016 13:59:25] "GET / HTTP/1.1" 304 -

Am I missing something?

/env/bin/pip list
docker-compose (1.5.2)
docker-py (1.6.0)
dockerpty (0.3.4)
docopt (0.6.2)
enum34 (1.1.2)
Flask (0.10.1)
functools32 (3.2.3.post2)
itsdangerous (0.24)
Jinja2 (2.8)
jsonschema (2.5.1)
MarkupSafe (0.23)
pip (7.1.2)
PyYAML (3.11)
requests (2.7.0)
setuptools (18.2)
six (1.10.0)
texttable (0.8.4)
websocket-client (0.35.0)
Werkzeug (0.11.3)
wheel (0.24.0)

**Comment by francescou** [comment]
Can you please tell me what you see in the container logs after accessing the application? Normally it should be something like this (notice the /api/v1... requests)

```
INFO:werkzeug:172.17.0.1 - - [06/Jan/2016 16:06:29] "GET / HTTP/1.1" 200 -
INFO:werkzeug:172.17.0.1 - - [06/Jan/2016 16:06:30] "GET /images/logo.png HTTP/1.1" 200 -
INFO:werkzeug:172.17.0.1 - - [06/Jan/2016 16:06:30] "GET /styles/colors.css HTTP/1.1" 200 -
...
INFO:werkzeug:172.17.0.1 - - [06/Jan/2016 16:06:30] "GET /scripts/services/logs.js HTTP/1.1" 200 -
INFO:werkzeug:172.17.0.1 - - [06/Jan/2016 16:06:30] "GET /api/v1/projects HTTP/1.1" 200 -
INFO:werkzeug:172.17.0.1 - - [06/Jan/2016 16:06:30] "GET /views/home.html HTTP/1.1" 200 -
INFO:werkzeug:172.17.0.1 - - [06/Jan/2016 16:06:30] "GET /api/v1/host HTTP/1.1" 200 -
```

If your logs looks different could you please check if there are issues on the browser? (on chrome F12 -> Console).

Otherwise, can you try to execute (on a clean working copy):

```
docker build -t foo/docker-compose-ui .
docker run --rm -p 6000:5000 -v /var/run/docker.sock:/var/run/docker.sock foo/docker-compose-ui
```

and then open http://localhost:5001/#/ in your browser

Thank you.

Francesco

**Comment by urban-it-solutions** [comment]
Hi Francesco,

it´s stuck at

INFO:werkzeug:172.17.0.1 - - [06/Jan/2016 16:06:29] "GET / HTTP/1.1" 200 -

It does not work with Chrome (v 47) but with Firefox...

Thank you!

Axel

**Comment by francescou** [comment]
ok, so it's a client-side error. Could you please check the error logs on the [chrome devtools console](https://developer.chrome.com/devtools/docs/console) ?

**Comment by urban-it-solutions** [comment]
Now I understand! Its not a Browser problem...the application is connecting to rawgithub, cloudflare etc
I had no Internet-Connection in my Chrome-Browser...
Is it possible to integrate angular.js, css, jquery etc inside the container?

**Comment by francescou** [comment]
yes it could be done using [bower](http://bower.io/#install-bower).
I've created an [experimental branch: "bower"](https://github.com/francescou/docker-compose-ui/tree/bower)

If you want to try install bower and then run:

```
git clone git@github.com:francescou/docker-compose-ui.git
cd docker-compose-ui
git checkout bower
(cd static/ && bower install)
docker build -t docker-compose-ui-bower .
docker run --rm -p 5003:5000 -v /var/run/docker.sock:/var/run/docker.sock docker-compose-ui-bower 
```

**Comment by urban-it-solutions** [comment]
great! will try that.

I have another issue...not a big thing.

Here is my compose-file:

sles12-elk-malte:
    image: 172.20.132.109:5000/sles12-elk
    container_name: sles12-elk-malte
    cpu_shares: 1024
    cpuset: 0,0
    hostname: elktest
    domainname: int.hlg.de
    mem_limit: 2g
    memswap_limit: 2g

```
ports:
    - "6380:6379"
    - "127.0.0.1:5602:5601"

volumes:
    - "/opt/elk/malte/conf:/conf"
    - "/opt/elk/malte/data:/data"
    - "/opt/elk/malte/logs:/logs"
```

But the GUI shows me:

 Network

localhost:5602 : 5601/tcp

localhost:6380 : 6379/tcp

The second binding to localhost is wrong...

cheers

Axel

**Comment by francescou** [comment]
I can't see the problem here.
Internal port 6379 is published on localhost:6380, right?
Maybe your problem is that it is not an http service but, say, a redis instance?

**Comment by urban-it-solutions** [comment]
no, internal port 6379 is published to 0.0.0.0:6380....

**Comment by urban-it-solutions** [comment]
e5af8d7502dc        172.20.132.109:5000/sles12-elk          "/usr/bin/supervisord"   12 seconds ago      Up 10 seconds       80/tcp, 9200/tcp, 9300/tcp, 127.0.0.1:5602->5601/tcp, 0.0.0.0:6380->6379/tcp   sles12-elk-malte

**Comment by francescou** [comment]
yes you're right, I've fixed the issue in 155a786
you can try the fix using the docker image `francescou/docker-compose-ui:latest`

**Comment by francescou** [comment]
I will close the issue, feel free to reopen if necessary.
